src/Microsoft.Health.Fhir.TemplateManagement.FunctionalTests/TemplateCollectionFunctionalTests.cs

```diff
@@ -264,6 +273,25 @@ public async Task GiveDefaultImageReference_WhenGetTemplateCollectionWithEmptyTo
             Assert.Equal(defaultTemplatesCounts, templateCollection.First().Count());
         }
 
+        // Conversion results of DefaultTemplates.tar.gz and default template folder should be the same.
+        [Theory]
+        [MemberData(nameof(GetHl7v2DataAndTemplateSources))]
+        public async Task GivenSameInputData_WithDifferentTemplateSource_WhenConvert_ResultShouldBeIdentical(string inputFile, string defaultTemplateDirectory, string rootTemplate)
+        {
+            var folderTemplateProvider = new Hl7v2TemplateProvider(defaultTemplateDirectory);
+
+            var templateProviderFactory = new TemplateCollectionProviderFactory(new MemoryCache(new MemoryCacheOptions()), Options.Create(new TemplateCollectionConfiguration()));
+            var templateProvider = templateProviderFactory.CreateTemplateCollectionProvider(ImageInfo.DefaultTemplateImageReference, string.Empty);
+            var imageTemplateProvider = new Hl7v2TemplateProvider(await templateProvider.GetTemplateCollectionAsync(CancellationToken.None));
+
+            var hl7v2Processor = new Hl7v2Processor();
+            var inputContent = File.ReadAllText(inputFile);
+
+            var imageResult = hl7v2Processor.Convert(inputContent, rootTemplate, imageTemplateProvider);
+            var folderResult = hl7v2Processor.Convert(inputContent, rootTemplate, folderTemplateProvider);
+            Assert.Equal(imageResult, folderResult);
```

Comment:
These two strings are JSON objects actually, can we compare them by string.comparison?

Reply:
Update: Currently we are using a lot of this kind of comparison methods, but only in testing. I think it's ok.

Reply:
Is it possible that same inputdata will be serialized to different json string? I feel the serialization result should be stable